### DISCUSSION TITLE
SetDns method manually

### DISCUSSION THREAD

**Discussion by BrotherV** [esp8266/Arduino#8477] [Ideas]
### Description

I have a problem with WiFiClientSecure, it can't make a connection to HTTPS and HTTP servers (port 443 and 80). I know that I have to set the DNS server IP, however, since we have many nodes that connect to a router, dedicating a static IP for each one is not a good solution. All of them have the same code and the IP must be set dynamically by the router. The point is, I want to set the DNS server IP manually but there is not such a command for this purpose. 
I've read about the Arduino WiFi shield and it has a command to set one or two DNS servers without setting static IP. I want to know is it possible to add this command to the library?!!

```cpp

 WiFi.setDNS(dns1);     or    WiFi.setDNS(dns1, dns2); 

```

### Example

```cpp

#include <Arduino.h>

// Network Configurations
#define WLAN_SSID     "YOUR_WIRELESS_SSID"
#define WLAN_PASSWORD "YOUR_PASSWORD"

// the IP address for the shield:
IPAddress dns1(8, 8, 8, 8);  //Google dns
IPAddress dns2(8, 8, 4, 4);  //Alternate dns

void setup() {

    WiFi.mode(WIFI_STA);
    WiFi.begin(WLAN_SSID, WLAN_PASSWORD);
    while (WiFi.status() != WL_CONNECTED) {
      delay(500);
      Serial.print(".");
    }
  
    Serial.println();
    Serial.println("WiFi connected");
    Serial.println("IP address: ");
    Serial.println(WiFi.localIP());

    // print your WiFi IP address:
    WiFi.setDNS(dns1);              //WiFi.setDNS(dns1, dns2); 
    Serial.print("Dns configured.");

}

void loop() {}

```

**Comment by d-a-v** [comment]
Use [`dns_setserver()`](http://www.nongnu.org/lwip/2_0_x/group__dns.html), The max number of dns server is [there](https://github.com/esp8266/Arduino/blob/9f536e68f8f3aa3ffae26493dd0a31e1957d2e81/tools/sdk/lwip2/include/lwipopts.h#L1120-L1126).
Best would be that your router itself give the DNS server, as it always should happen.
Be aware that at the end of each lease time, dhcp client (esp) re-request an IP from the ~~router~~dhcp-server,
if there is a DNS misconfiguration in the router, your setting may be overwritten.
